### PR TITLE
[develop]: Document process for changing number of vertical levels

### PR DESCRIPTION
**Pull request by gspetro-NOAA**
## DESCRIPTION OF CHANGES: 
This PR adds a section to the LAMGrid chapter explaining how to change the number of vertical levels in an SRW App experiment 

### Type of change
- [X] This change requires a documentation update

## TESTS CONDUCTED: 
None required. This is a documentation-only PR. However, I ran through an experiment using the process in the docs to generate ak/bk for 128 levels. I used the resulting VCOORD_FILE to run the `community.config.yaml` experiment with 128 vertical levels instead of the usual 65. The workflow completed successfully and the results can be viewed on Hera here: `/scratch2/NAGAPE/epic/Gillian.Petro/expt_dirs/test_vert2`
The human-friendly version of the docs in my fork can be viewed here: https://srw-ug.readthedocs.io/en/text-grid/CustomizingTheWorkflow/LAMGrids.html.

- [ ] hera.intel
- [ ] orion.intel
- [ ] cheyenne.intel
- [ ] cheyenne.gnu
- [ ] gaea.intel
- [ ] jet.intel
- [ ] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [ ] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)

## DEPENDENCIES:
PR #880 

## DOCUMENTATION:
N/A (all docs)

## ISSUE: 
It seems that PR #813 and the updates in this PR at least partially resolve Issue #397. 
I think there was the hope that we could simplify the process for users (e.g., by setting `levp` / `npz` and other required variables in `config.yaml` and generating the ak/bk automatically from that), so it may be best to check with @JeffBeck-NOAA and @gsketefian as to whether to close this issue and open an enhancement issue or just keep this issue open. 

## CHECKLIST
- [x] My code follows the style guidelines in the Contributor's Guide
- [X] I have performed a self-review of my own code using the Code Reviewer's Guide
- [ ] I have commented my code, particularly in hard-to-understand areas (N/A)
- [X] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain).
- [X] My changes generate no new warnings
- [ ] New and existing tests pass with my changes (N/A)
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

File: docs/UsersGuide/source/CustomizingTheWorkflow/LAMGrids.rst [hunk 1]
```diff
@@ -49,12 +49,12 @@ These four options are provided for flexibility related to compute resources and
...
 
-The predefined :term:`CONUS` grids follow the naming convention (e.g., RRFS_CONUS_*km) of the prototype 3-km continental United States (CONUS) grid being tested for the Rapid Refresh Forecast System (:term:`RRFS`). The RRFS will be a convection-allowing, hourly-cycled, :term:`FV3`-:term:`LAM`-based ensemble planned for operational implementation in 2024. All four supported grids were created to fit completely within the High Resolution Rapid Refresh (`HRRR <https://rapidrefresh.noaa.gov/hrrr/>`_) domain to allow for use of HRRR data to initialize the SRW App. 
+The predefined :term:`CONUS` grids follow the naming convention (e.g., ``RRFS_CONUS_*km``) of the prototype 3-km continental United States (CONUS) grid being tested for the Rapid Refresh Forecast System (:term:`RRFS`). RRFS will be a convection-allowing, hourly-cycled, :term:`FV3`-:term:`LAM`-based ensemble planned for operational implementation in late 2024. All four supported grids were created to fit completely within the High Resolution Rapid Refresh (`HRRR <https://rapidrefresh.noaa.gov/hrrr/>`_) domain to allow for use of HRRR data to initialize the SRW App.
```
Comment (JeffBeck-NOAA) [inline docs/UsersGuide/source/CustomizingTheWorkflow/LAMGrids.rst:53]: I would say "3-km version of the continental United States (CONUS) grid being tested for the RRFS".  The RRFS will use a North American domain, but has been tested on a CONUS domain for a while due to computational constraints.  Also, the timeline for RRFSv1 is now into 2025, so I'd just say "in 2025".

File: docs/UsersGuide/source/CustomizingTheWorkflow/LAMGrids.rst [hunk 2]
```diff
@@ -160,45 +162,260 @@ The following is an example of a code stanza for "NEW_GRID" to be added to ``pre
...
 
 .. note:: 
    The process above explains how to create a new *predefined* grid, which can be used more than once. If a user prefers to create a custom grid for one-time use, the variables above can instead be specified in ``config.yaml``, and ``PREDEF_GRID_NAME`` can be set to a null string. In this case, it is not necessary to modify ``valid_param_vals.yaml`` or ``predef_grid_params.yaml``. Users can view an example configuration file for a custom grid `here <https://github.com/ufs-community/ufs-srweather-app/blob/develop/tests/WE2E/test_configs/wflow_features/config.custom_ESGgrid.yaml>`__.
+
+.. _VerticalLevels:
+
+Changing the Number of Vertical Levels
+========================================
+
+The four supported predefined grids included with the SRW App have 65 vertical levels. However, advanced users may wish to vary the number of vertical levels in the grids they are using, whether these be the predefined grids or a user-generated grid. Varying the number of vertical levels requires
```
Comment (JeffBeck-NOAA) [inline docs/UsersGuide/source/CustomizingTheWorkflow/LAMGrids.rst]: I would say that the four grids "are configured to run with 65 levels by default".

File: docs/UsersGuide/source/CustomizingTheWorkflow/LAMGrids.rst [hunk 3]
```diff
@@ -160,45 +162,260 @@ The following is an example of a code stanza for "NEW_GRID" to be added to ``pre
...
+knowledge of how the SRW App interfaces with the UFS Weather Model (:term:`WM <Weather Model>`) and preprocessing utilities. It is also important to note that user-defined vertical levels are not a supported feature at present; information is being provided for the benefit of the FV3-LAM community, but user support for this feature is limited. With those caveats in mind, this section provides instructions for modifying the number of vertical levels on a regional grid. 
+
```
Comment (JeffBeck-NOAA) [inline docs/UsersGuide/source/CustomizingTheWorkflow/LAMGrids.rst:216]: "Modifying" or "creating a user-defined vertical coordinate distribution"?

File: docs/UsersGuide/source/CustomizingTheWorkflow/LAMGrids.rst [hunk 4]
```diff
@@ -160,45 +162,260 @@ The following is an example of a code stanza for "NEW_GRID" to be added to ``pre
...
+Users can find ``vcoord_gen`` `technical documentation here <https://noaa-emcufs-utils.readthedocs.io/en/latest/ufs_utils.html#vcoord-gen>`__ and `scientific documentation here <https://ufs-community.github.io/UFS_UTILS/vcoord_gen/vcoord__gen_8f90.html>`__. Since UFS_UTILS is part of the SRW App, users can find and run the UFS_UTILS ``vcoord_gen`` tool in their ``ufs-srweather-app/exec`` directory. To run ``vcoord_gen`` within the SRW App: 
+
+.. code-block:: console 
+
+   cd /path/to/ufs-srweather-app/exec
+   ./vcoord_gen > /path/to/vcoord_gen_outfile.txt
+
+Users should modify the output file path to save the output file in the desired location. In the SRW App, the default file defining vertical levels is named ``global_hyblev.txt``, so by convention, a file with a different number of levels is named according to its number of levels (e.g., ``global_hyblev.L128.txt``). Configuration files are typically placed in the ``parm`` directory. For example, users might run:
+
```
Comment (JeffBeck-NOAA) [inline docs/UsersGuide/source/CustomizingTheWorkflow/LAMGrids.rst:232]: I might say something like "by convention, this file contains the default 65 levels", and that when creating a new vertical coodinate distribution file, one may append this file name with LXX or LXXX for their number of levels".

**Review by JeffBeck-NOAA — APPROVED** [review]
Thanks for all your work on this PR, @gspetro-NOAA!

**Review by MichaelLueken — APPROVED** [review]
@gspetro-NOAA - Thank you very much for going into great detail on how to change the number of vertical levels in the SRW App!  These changes look good to me!  Approving now.

**Pull request merged**